### PR TITLE
Windows Snackbar rewrite #1658

### PR DESCRIPTION
**Pull request by VladislavAntonyuk**

### CODE REVIEW

**Review by brminnick — COMMENTED** [review]
Thanks Shaun! I took a stab making the new `Package.appxmanifest` code a bit more modular. Let me know what you think!

File: docs/maui/alerts/snackbar.md [hunk 1]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+
+No setup is required.
+
+### [Windows](#tab/windows)
+
+When using `Snackbar` it is essential to perform the following two steps:
+
+#### 1. Enable the snackbar usage with the MauiAppBuilder
+
+When using the `UseMauiCommunityToolkit` make use of the `options` parameter to enable the snackbar usage on windows as follows:
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:33]:
```suggestion
When using the `UseMauiCommunityToolkit` make use of the `options` parameter to enable the snackbar usage on Windows as follows:
```

File: docs/maui/alerts/snackbar.md [hunk 2]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+
+```csharp
+var builder = MauiApp.CreateBuilder()
+  .UseMauiCommunityToolkit(options =>
+  {
+    options.SetShouldEnableSnackbarOnWindows(true);
+  })
+```
+
+The above will automatically register the required handlers by configuring lifecycle events (OnLaunched and OnClosed).
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:43]:
```suggestion
The above will automatically register the required handlers by configuring lifecycle events (`OnLaunched` and `OnClosed`).
```

File: docs/maui/alerts/snackbar.md [hunk 3]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10"
+```
+
+2. In **Package.appxmanifest**, also in the opening `<Package>` tag, update `IgnorableNamespaces` to include `uap` `rescap` `com` and `desktop`:
+
+```xml
+IgnorableNamespaces="uap rescap com desktop"
+```
+
+##### Example:Completed `<Package>` Tag
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:62]:
```suggestion
##### Example: Completed `<Package>` Tag
```

File: docs/maui/alerts/snackbar.md [hunk 4]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+     xmlns:desktop="http://schemas.microsoft.com/appx/manifest/desktop/windows10"
+     IgnorableNamespaces="uap rescap com desktop">
+```
+
+3. In **Package.appxmanifest**, inside of each `<Application>` tag, add the following extensions:
+
+```xml
+<Extensions>
+
+  <!--Specify which CLSID to activate when notification is clicked-->
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:81]:
```suggestion
   <!-- Specify which CLSID to activate when notification is clicked -->
```

File: docs/maui/alerts/snackbar.md [hunk 5]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+   <desktop:Extension Category="windows.toastNotificationActivation">
+       <desktop:ToastNotificationActivation ToastActivatorCLSID="6e919706-2634-4d97-a93c-2213b2acc334" /> 
+   </desktop:Extension>
+
+   <!--Register COM CLSID-->
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:86]:
```suggestion
   <!-- Register COM CLSID -->
```

File: docs/maui/alerts/snackbar.md [hunk 6]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+   <com:Extension Category="windows.comServer">
+       <com:ComServer>
+           <com:ExeServer Executable="YOUR-PATH-TO-EXECUTABLE" DisplayName="$targetnametoken$" Arguments="----AppNotificationActivated:"> <!--Example path to executable: CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.exe-->
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:89]:
```suggestion
           <com:ExeServer Executable="YOUR-PATH-TO-EXECUTABLE" DisplayName="$targetnametoken$" Arguments="----AppNotificationActivated:"> <!-- Example path to executable: CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.exe -->
```

File: docs/maui/alerts/snackbar.md [hunk 7]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+       </uap:VisualElements>
+       <Extensions>
+
+         <desktop:Extension Category="windows.toastNotificationActivation">
+             <desktop:ToastNotificationActivation ToastActivatorCLSID="6e919706-2634-4d97-a93c-2213b2acc334" /> 
+         </desktop:Extension>
+      
+         <com:Extension Category="windows.comServer">
+             <com:ComServer>
+                 <com:ExeServer Executable="YOUR-PATH-TO-EXECUTABLE" DisplayName="$targetnametoken$" Arguments="----AppNotificationActivated:"> <!--Example path to executable: CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.exe-->
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:122]:
```suggestion
                 <com:ExeServer Executable="YOUR-PATH-TO-EXECUTABLE" DisplayName="$targetnametoken$" Arguments="----AppNotificationActivated:"> <!-- Example path to executable: CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.exe -->
```

File: docs/maui/alerts/snackbar.md [hunk 8]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+           </com:ExeServer>
+       </com:ComServer>
+   </com:Extension>
+
+</Extensions>
+```
+
+##### Example: Completed `<Applications>` tag
+
+Here is an example of a completed `<Applications>` tag that now has added support for `Snackbar` ***:
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:100]:
```suggestion
Here is an example of a completed `<Applications>` tag that now has added support for `Snackbar`:
```

File: docs/maui/alerts/snackbar.md [hunk 9]
```diff
@@ -11,6 +11,201 @@ The `Snackbar` is a timed alert that appears at the bottom of the screen by defa
...
+            </uap:VisualElements>
+            <Extensions>
+
+               <desktop:Extension Category="windows.toastNotificationActivation">
+                   <desktop:ToastNotificationActivation ToastActivatorCLSID="6e919706-2634-4d97-a93c-2213b2acc334" /> 
+               </desktop:Extension>
+            
+               <com:Extension Category="windows.comServer">
+                   <com:ComServer>
+                       <com:ExeServer Executable="YOUR-PATH-TO-EXECUTABLE" DisplayName="$targetnametoken$" Arguments="----AppNotificationActivated:"> <!--Example path to executable: CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.exe-->
```
Comment (jfversluis) [inline docs/maui/alerts/snackbar.md:183]:
```suggestion
                       <com:ExeServer Executable="YOUR-PATH-TO-EXECUTABLE" DisplayName="$targetnametoken$" Arguments="----AppNotificationActivated:"> <!-- Example path to executable: CommunityToolkit.Maui.Sample\CommunityToolkit.Maui.Sample.exe -->
```

**Review by jfversluis — APPROVED** [review]

**Pull request merged**